### PR TITLE
Notify users when a declare is used

### PR DESCRIPTION
**Pull request by HoeenCoder**
Client pull request: Zarel/Pokemon-Showdown-Client#873

Check the client pull request for details on these changes.

---

Previous Pull Request was closed because I managed to screw up my repo, these are the same changes as before.

### CODE REVIEW

File: chat-commands.js [hunk 1]
```diff
@@ -1990,7 +1990,8 @@ exports.commands = {
 		if (!this.can('declare', null, room)) return false;
 		if (!this.canTalk()) return;
 
-		this.add('|raw|<div class="broadcast-blue"><b>' + Chat.escapeHTML(target) + '</b></div>');
+		this.add(`|notify|The staff in ${room.id} made a declaration|${Chat.escapeHTML(target)}`);
```
Comment (Zarel) [inline chat-commands.js]: Do notifications support HTML?

Comment (HoeenCoder) [inline chat-commands.js]: From the tests ive done, no.

Comment (Zarel) [inline chat-commands.js]: In which case we probably want to strip HTML from `target`. Maybe introduce a `Chat.stripHTML`.

Comment (Zarel) [inline chat-commands.js]: And iirc notification titles are usually are really short. So this might just be `${room.title} announcement!`

Comment (HoeenCoder) [inline chat-commands.js]: I need to learn regex to do that, good reason to finally do it lol.

Sure, I can change the title.

**Comment by HoeenCoder** [comment]
After poking around a bit with regex (and google) I found this which seems to work
`str.replace(/<(?:.|\n)*?>/gm, '');` Which removes angle brackets and everything in between except newlines. Would this be enough for a stripHTML function? or is it too much/too little @Zarel ?

File: chat-commands.js (hunk 1 above)
Comment (Zarel) [inline chat-commands.js]:
```
stripHTML(html) {
    return html.replace(/<[^>]*>/g, '');
}
```

Maybe also replace entities, though.

**Comment by Zarel** [comment]
That's... one way to do it. Usually I prefer `[^>]*` to `.*?`.

**Comment by HoeenCoder** [comment]
Yours is alot easier to understand too, i'll use that.

**Comment by HoeenCoder** [comment]
I borrowed DeathlyPlays C9 to rebase btw, thats why hes showing up on the commits

File: chat-commands.js [hunk 2]
```diff
@@ -1990,7 +1990,8 @@ exports.commands = {
 		if (!this.can('declare', null, room)) return false;
 		if (!this.canTalk()) return;
 
-		this.add('|raw|<div class="broadcast-blue"><b>' + Chat.escapeHTML(target) + '</b></div>');
+		this.add(`|notify|${room.id} announcement!|${Chat.stripHTML(target)}`);
```
Comment (Zarel) [inline chat-commands.js]: `room.title` makes more sense here, I think.

File: chat-commands.js [hunk 3]
```diff
@@ -1990,7 +1990,8 @@ exports.commands = {
 		if (!this.can('declare', null, room)) return false;
 		if (!this.canTalk()) return;
 
-		this.add('|raw|<div class="broadcast-blue"><b>' + Chat.escapeHTML(target) + '</b></div>');
+		this.add(`|notify|${room.title} announcement!|${Chat.stripHTML(target)}`);
```
Comment (Zarel) [inline chat-commands.js]: `target` already has no HTML; it doesn't need to be escaped further.

File: chat-commands.js [hunk 4]
```diff
@@ -1990,7 +1990,8 @@ exports.commands = {
...
+		this.add(`|raw|<div class="broadcast-blue"><b>${Chat.escapeHTML(target)}</b></div>`);
```
Comment (Zarel) [inline chat-commands.js]: `Chat.html`

File: chat-commands.js [hunk 5]
```diff
@@ -2029,7 +2034,10 @@ exports.commands = {
 		if (!target) return;
 
 		Rooms.rooms.forEach((curRoom, id) => {
-			if (id !== 'global' && curRoom.type !== 'battle') curRoom.addRaw('<div class="broadcast-blue"><b>' + target + '</b></div>').update();
+			if (id !== 'global' && curRoom.type !== 'battle') curRoom.addRaw(`<div class="broadcast-blue"><b>${target}</b></div>`).update();
+		});
+		Users.users.forEach(u => {
+			if (u.connected) u.send(`|pm|~|${u.group}${u.name}|/raw <div class="broadcast-blue"><b>${target}</b></div>`);
```
Comment (Zarel) [inline chat-commands.js]: `cdeclare` probably shouldn't send a PM.

What are the use cases of `cdeclare`, anyway?

Comment (Zarel) [inline chat-commands.js]: (On the other hand, lockdowns probably should PM.)

**Comment by Zarel** [comment]
If you're not going to maintain this, I guess I have to...

**Comment by HoeenCoder** [comment]
I wasnt aware that there were any concerns about this, I guess github never sent me a notification that you reviewed...

**Pull request merged**